Makefile

```diff
@@ -369,3 +369,41 @@ multimod-verify: install-tools
 multimod-prerelease: install-tools
 	multimod prerelease -v ./versions.yaml -m collector-core
 	$(MAKE) gotidy
+
+.PHONY: prepare-release
+prepare-release:
+ifdef PREVIOUS_VERSION
+	@echo "Previous version $(PREVIOUS_VERSION)"
+else
+	@echo "PREVIOUS_VERSION not defined"
+	@echo "usage: make prepare-release RELEASE_CANDIDATE=<version eg 0.53.0> PREVIOUS_VERSION=<version eg 0.52.0>"
+	exit 1
+endif
+ifdef RELEASE_CANDIDATE
+	@echo "Preparing release $(RELEASE_CANDIDATE)"
+else
+	@echo "RELEASE_CANDIDATE not defined"
+	@echo "usage: make prepare-release RELEASE_CANDIDATE=<version eg 0.53.0> PREVIOUS_VERSION=<version eg 0.52.0>"
+	exit 1
+endif
+	# ensure a clean branch
+	git diff -s --exit-code || (echo "local repository not clean"; exit 1)
+	# TODO: update changelog
+	# update versions.yaml config.go builder-config.yaml
+	sed -i.bak 's/$(PREVIOUS_VERSION)/$(RELEASE_CANDIDATE)/g' versions.yaml
+	sed -i.bak 's/$(PREVIOUS_VERSION)/$(RELEASE_CANDIDATE)/g' ./cmd/builder/internal/builder/config.go
+	sed -i.bak 's/$(PREVIOUS_VERSION)/$(RELEASE_CANDIDATE)/g' ./cmd/otelcorecol/builder-config.yaml
+	sed -i.bak 's/$(PREVIOUS_VERSION)/$(RELEASE_CANDIDATE)/g' examples/k8s/otel-config.yaml
+	find . -name "*.bak" -type f -delete
+	# regenerate files
+	$(MAKE) genotelcorecol
+	# commit changes before running multimod
+	git checkout -b opentelemetry-collector-bot/release-$(RELEASE_CANDIDATE)
+	git add .
+	git commit -m "prepare release $(RELEASE_CANDIDATE)"
+	$(MAKE) multimod-prerelease
+	# commit multimod changes
+	git add .
+	git commit -m "add multimod changes" || (echo "no multimod changes to commit")
+	git push fork opentelemetry-collector-bot/release-$(RELEASE_CANDIDATE)
+	gh pr create --title "[chore] prepare release $(RELEASE_CANDIDATE)"
```

Comment:
This requires "gh" command, maybe just push and let everyone create PR manually?

Reply:
Or check whether `gh` exists and use it, otherwise tell the user "I could have done your job for you, but you didn't let me"? I think we have something similar in the operator repository.

Reply:
That was actually for jaeger's: https://github.com/jaegertracing/jaeger-operator/blob/3497c5335b70e4a0687d14445d02319c70f67170/.ci/operatorhub.sh#L47-L51

Reply:
Added a check for `gh`